src/components/videoCard/videoCard.jsx

```diff
@@ -1,11 +1,23 @@
 import "../styles/main.css";
 import { useVideo } from "../../context/videoContext";
+import { useWatchLater } from "../../context/watchLaterContext";
+import { useState } from "react";
 
 const VideoCard = (props) => {
+  const [watchLaterClicked, setWatchLaterClicked] = useState(false);
   const videoData = props.video;
-  console.log(videoData);
   const { videoList } = useVideo();
-  console.log(videoList);
+  const { watchLaterDispatch } = useWatchLater();
+  
+  const watchLaterHandler = () => {
+    watchLaterClicked ? setWatchLaterClicked(false) : setWatchLaterClicked(true);
+    watchLaterDispatch({
+                 type: "ADD_VIDEO_TO_WATCH_LATER",
+                 payload: videoData,
+               })
```

Comment:
With this logic, if I click multiple times on the the watchlater icon, that video will get added multiple times to the watch later videos list. You can give a condition, if the video is already in the watch later list(or if icon is already clicked) then the ```watchLaterDispatch``` should be called with type ```"REMOVE_VIDEO_FROM_WATCH_LATER"```.
Attaching a screenshot for your reference
<img width="949" alt="image" src="https://user-images.githubusercontent.com/64582473/161387871-dca16605-27f4-4bb4-8d96-1d0eb5fe2da0.png">

Reply:
Thank you for your valuable feedback :
Will definitely make the suggested changes.